com.unity.cinemachine/Runtime/Behaviours/CinemachineStateDrivenCamera.cs

```diff
@@ -131,7 +131,7 @@ protected internal override void LegacyUpgrade(int streamedVersion)
             {
                 DefaultTarget = new DefaultTargetSettings 
                 { 
-                    Enabled = m_LegacyLookAt != null || m_LegacyFollow != null,
```

Comment:
Why this change?

Reply:
Fixes: [CMCL-1157](https://jira.unity3d.com/browse/CMCL-1157) - we may need to come up with something else. LegacyUpgrade cannot check for null, because that is only allowed on the main thread. However, OnAfterDeserialize is not guaranteed to run on the main thread. When it does not run on main thread we get an error that stops LegacyUpgrade.
(Error message about EnsureRunningOnMainThread)

Reply:
I am confused.  What error was happening, and when was it happening?  The Jira ticket doesn't say

Reply:
The error happens when LegacyUpgrade is **not** called from the main thread. 

OnAfterDeserialize calls LegacyUpgrade. Unfortunately, OnAfterDeserialize is not guaranteed to be called from the main thread.

Reply:
Can you be more specific?  Where is the offending null check?

Reply:
nvm I understand now :)

Reply:
Let's pull this out into a separate PR.  I think we need a different fix.

Reply:
https://github.com/Unity-Technologies/com.unity.cinemachine/pull/534